### ISSUE TITLE
Plugin "shell-proxy" error: proxy:20: no such file or directory: ./proxy.py

### ISSUE THREAD

**Issue by shelfofclub** [ohmyzsh/ohmyzsh#10858]
### Describe the bug

Plugin "shell-proxy" doesn't work. It said "proxy:20: no such file or directory: ./proxy.py".

Line 20 is `  local proxy="${functions_source[$0]:A:h}/proxy.py"`. `echo ${functions_source[proxy]:A:h}` tells me `.`.

The reason may be that zsh is too old in centos7. It works in ubuntu 20.04 with zsh 5.8

### Steps to reproduce

1. Enable plugin "shell-proxy".
2. Run command `proxy --help`.
3. Error is "proxy:20: no such file or directory: ./proxy.py".

### Expected behavior

Plugin "shell-proxy" should work well.

`proxy --help` may output:
```
shell-proxy: no proxy configuration found.

Set `SHELLPROXY_URL` or create a config file at `/home/myname/.config/proxy`
See the plugin README for more information.
```

### OS / Linux distribution

CentOS Linux release 7.9.2009 (Core)

### Zsh version

5.0.2

### Oh My Zsh version

master (30e23a6)

### Terminal emulator

Windows Terminal 1.9.1942.0

**Comment by mcornella** [comment]
Can you run `echo ${functions_source[proxy]}`?

**Comment by shelfofclub** [comment]
It seems to output nothing.

```zsh
$ echo ${functions_source[proxy]}

$ result=${functions_source[proxy]}
$ echo ${#result}
0
```

And as you suggested in #9558 : 
```zsh
$ whence -c proxy
proxy () {
        if [[ -n "$DEFAULT_PROXY" && -z "$SHELLPROXY_URL" ]]
        then
                echo "proxy: DEFAULT_PROXY is deprecated, use SHELLPROXY_URL instead" >&2
                SHELLPROXY_URL="$DEFAULT_PROXY"
                unset DEFAULT_PROXY
        fi
        if [[ -n "$CONFIG_PROXY" && -z "$SHELLPROXY_CONFIG" ]]
        then
                echo "proxy: CONFIG_PROXY is deprecated, use SHELLPROXY_CONFIG instead" >&2
                SHELLPROXY_CONFIG="$CONFIG_PROXY"
                unset CONFIG_PROXY
        fi
        local proxy="${functions_source[$0]:A:h}/proxy.py"
        local output
        output="$(SHELLPROXY_URL="$SHELLPROXY_URL" SHELLPROXY_CONFIG="$SHELLPROXY_CONFIG" "$proxy" "$1")"  || return $?
        source <(echo "$output")
}

$ functions -t proxy
$ proxy --help
+proxy:2> [[ -n '' ]]
+proxy:9> [[ -n '' ]]
+proxy:16> local 'proxy=./proxy.py'
+proxy:19> local output
+proxy:20> output=+proxy:20> SHELLPROXY_URL='' SHELLPROXY_CONFIG='' ./proxy.py --help
proxy:20: no such file or directory: ./proxy.py
+proxy:20> output=''
+proxy:21> return 127
```

**Comment by mcornella** [comment]
Sorry, I misread the initial post. Did you install Oh My Zsh normally or are you using a plugin manager? What does `zmodload` show?

**Comment by shelfofclub** [comment]
I installed omz by install.sh and I don't use other plugin manager, just zsh and omz.
zmodload shows: 
```
zsh/complete
zsh/complist
zsh/computil
zsh/datetime
zsh/langinfo
zsh/main
zsh/parameter
zsh/regex
zsh/stat
zsh/terminfo
zsh/zle
zsh/zleparameter
zsh/zutil

I use these plugins:
plugins=(tmux conda-zsh-completion zsh-syntax-highlighting nvm extract colored-man-pages shell-proxy)
```
On ubuntu 20.04 where shell-proxy works well, the output is:
```zsh
~ » echo ${functions_source[proxy]}
/home/myname/.oh-my-zsh/plugins/shell-proxy/shell-proxy.plugin.zsh

~ » zmodload
zsh/complete
zsh/complist
zsh/datetime
zsh/langinfo
zsh/main
zsh/parameter
zsh/regex
zsh/stat
zsh/terminfo
zsh/zle
zsh/zleparameter
zsh/zutil

I use these plugins on ubuntu:
plugins=(tmux conda-zsh-completion zsh-syntax-highlighting extract colored-man-pages shell-proxy)
```

**Comment by shelfofclub** [comment]
Oh!!! There is no `functions_source` on centos7 with zsh 5.0.2!

`echo ${functions_source}` outputs nothing.

`functions_source` came with zsh >=5.4. [zsh news](https://github.com/zsh-users/zsh/blob/00d20ed15e18f5af682f0daec140d6b8383c479a/NEWS#L206)

**Comment by mcornella** [comment]
Oh right, that explains it then. We're using it in other places, I'll make sure to update them. Thanks for the discovery!

**Comment by mcornella** [comment]
Fixed in 4fcf52c2b1e086596d96b04f9c1ff3d742194857. Thanks for the report!

**Comment by shelfofclub** [comment]
@mcornella Thanks!
